lib/class-wp-theme-json-gutenberg.php

```diff
@@ -991,7 +991,7 @@ protected static function get_blocks_metadata() {
 		}
 
 		foreach ( $blocks as $block_name => $block_type ) {
-			$root_selector = wp_get_block_css_selector( $block_type );
+			$root_selector = ':where(' . wp_get_block_css_selector( $block_type ) . ')';
```

Comment:
This approach would need to be extended to where all the feature selectors are generated e.g. an image block's border styles.